server/tserver/src/main/java/org/apache/accumulo/tserver/log/LogSorter.java

```diff
@@ -292,11 +289,27 @@ void writeBuffer(String destPath, List<Pair<LogFileKey,LogFileValue>> buffer, in
     }
   }
 
-  public void startWatchingForRecoveryLogs(ThreadPoolExecutor distWorkQThreadPool)
+  /**
+   * Sort any logs that need sorting in the current thread.
+   */
+  public void sortLogsIfNeeded() throws KeeperException, InterruptedException {
+    new DistributedWorkQueue(context.getZooKeeperRoot() + Constants.ZRECOVERY, sortedLogConf,
```

Comment:
This method was added for the Compactor. Instead of the Compactor calling the other method that starts a background thread to watch for future log sorting work, it calls this method to process any log sorting work before asking for the next compaction job. In this case I don't think we want to wait, we want to execute immediate any log sorting work in the Compactor thread.

Reply:
Was not thinking of waiting, was just thinking of making the rate at which ZK is polled for recovery work consistent across server types.  Something like the following.

```suggestion
   if((currTime - lastCheckTime) > checkThreshold){
    // the checkThreshold exists to so that the compactor polls ZK for recovery work at a rate that is similar to tservers and sservers
    new DistributedWorkQueue(context.getZooKeeperRoot() + Constants.ZRECOVERY, sortedLogConf,....);
    lastCheckTime = currTime;
    }
```

Reply:
Those values are only used when it creates the background thread in `processExistingAndFuture`. The TabletServer and ScanServer call that method. The Compactor calls `processExistingWork` which does not set up a background thread to poll ZK for work.

Reply:
I realize the implementations are very different.  Was just thinking the different impls could possibly have same minimum poll rate for ZK